lms/services/__init__.py

```diff
@@ -109,9 +108,7 @@ def includeme(config):
     config.register_service_factory(
         "lms.services.ltia_http.factory", iface=LTIAHTTPService
     )
-    config.register_service_factory(
-        "lms.services.mailchimp.factory", iface=MailchimpService
-    )
+    config.register_service_factory("lms.services.mailchimp.factory", name="mailchimp")
```

Comment:
I changed this to name-based service registration to avoid a circular import

Reply:
You can use both methods, that won't work if the problem is not in the caller side but here.